composer/tools/composer_db_transfer.md

```diff
@@ -0,0 +1,160 @@
+
+## Limitations
+
+1.  Logs from old DAG executions are not transferred (but history of executions
+    in the database is transferred).
+
+1.  The script does not perform any DAG modifications, does not validate that
+    the DAGs are suitable to run in Airflow 2.0 and does not verify that
+    suitable dependencies are installed in target environment. Airflow
+    [upgrade_check](https://airflow.apache.org/docs/apache-airflow/stable/upgrade-check.html)
+    might be used to troubleshoot DAG compatibility issues.
+
+1.  `dag_code` and `serialized_dag` tables are not synchronized directly.
+    Database entries for imported DAGs are created by Airflow in the new
+    environment.
+
+1.  The script does not prevent DAGs from being scheduled in both old and new
+    environment. If can be prevented e.g. by pausing the DAGs before exporting
+    them and unpausing them in the new environment.
+
+1.  Airflow web server in target environment might temporarily not be available
+    after the database transfer.
+
+1.  The environment might need some time to parse imported DAGs. As a result
+    they might appear in the web UI with some delay.
+
+1.  Import operation removes all records from the target environment. Files
+    existing in environment's bucket are not deleted.
+
+1.  The script does not override gcloud configuration and exports it's own
+    kubeconfig. Temporary files for GKE deployment and kubeconfig created in
+    working directory have unique names.
+
+1.  File with fernet key created by export operation is not deleted by import
+    operation.
+
+## Usage
+
+### Export from an old environment (Airflow 1.10.14/15)
+
+CSV files with exported database are stored as /export/tables/[TABLE NAME].csv
+in the environment's bucket. dags, plugins and data directories are stored in
+/export/dirs in the environment's bucket.
+
+File with a fernet key is going to be created on the machine executing the
+script.
+
+```
+python3 composer_db_transfer.py export \
+    --project [PROJECT NAME] \
+    --environment [ENVIRONMENT NAME] \
+    --location [REGION] \
+    --fernet-key-file [PATH TO FERNET KEY FILE - TO BE CREATED]
+```
+
+### Copying the files between the environments
+
+Exported files can be copied to the target environment, e.g. with:
+
+```
+gsutil -m cp -r gs://[SOURCE ENV BUCKET NAME]/export gs://[TARGET ENV BUCKET NAME]/import
+```
+
+### Import to a new environment (Airflow 2.0.1+)
+
+CSV files that should be imported are expected to be stored as
+/import/tables/[TABLE NAME].csv in the environment's bucket. dags, plugins and
+data directories that should be imported are expected to be stored in
+/import/dirs in the environment's bucket.
+
+`fernet-key-file` parameter specifies path to the fernet key file of the source
+environment on the machine executing the script. It is created during export
+phase.
+
+```
+python3 composer_db_transfer.py import \
+    --project [PROJECT NAME] \
+    --environment [ENVIRONMENT NAME] \
```

Comment:
```suggestion
    --environment [NEW ENVIRONMENT NAME] \
```

Reply:
I want to avoid word "new" as it might suggest that the environment is created by import operation.
I'll use "source" and "target" words.